File: app/java-tdd-fizzbuzz/src/main/java/FizzBuzz.java
import java.util.ArrayList;
import java.util.List;

public class FizzBuzz {
    
    public String generate(int number) {
        if (number % 3 == 0 && number % 5 == 0) {
            return "FizzBuzz";
        } else if (number % 3 == 0) {
            return "Fizz";
        } else if (number % 5 == 0) {
            return "Buzz";
        }
        return String.valueOf(number);
    }
    
    public List<String> generateList(int count) {
        List<String> result = new ArrayList<>();
        for (int i = 1; i <= count; i++) {
            result.add(generate(i));
        }
        return result;
    }
    
    public void printFizzBuzz(int count) {
        List<String> result = generateList(count);
        for (String s : result) {
            System.out.println(s);
        }
    }
}

File: app/lib/fizz_buzz.rb
require_relative 'fizz_buzz/type'

class FizzBuzz
  MAX_NUMBER = 100
  attr_reader :list, :type

  def initialize(type)
    @type = type
  end

  def generate(number)
    @type.generate(number)
  end

  def generate_list
    @list = (1..MAX_NUMBER).map { |n| generate(n) }
  end
end
